server/src/routes/api/v1/voteRouter.js

```diff
@@ -0,0 +1,44 @@
+import express from "express";
+import { Review, Vote } from "../../../models/index.js";
+import { ValidationError } from "objection";
+import ReviewSerializer from "../../../serializers/ReviewSerializer.js"
+
+const voteRouter = new express.Router();
+
+voteRouter.post("/", async (req, res) => {
+    const voteValue = req.body.voteValue;
+    const reviewId = req.body.reviewId;
+    const userId = req.user.id;
+    try {
+        const voteExists = await Vote.query().where("reviewId",`${reviewId}`).where("userId",`${userId}`)
+        if (voteExists.length>0) {
+            if(voteValue === voteExists[0].voteValue) {
+                await Vote.query().delete().where("reviewId",`${reviewId}`).where("userId",`${userId}`)
+            } else { 
+                await Vote.query().delete().where("reviewId",`${reviewId}`).where("userId",`${userId}`)
+                const insertOppositeVote = await Vote.query().insertAndFetch({
+                    voteValue,
+                    userId,
+                    reviewId
+                });
```

Comment:
pretty sure we talked about how you would use patch or update here instead of these three queries!